Snail/Disposer.swift

```diff
@@ -8,15 +8,24 @@ public protocol DisposableType {
 
 public class Disposer {
     private(set) public var disposables: [DisposableType] = []
+    private let disposablesQueue = DispatchQueue(label: "snail-disposer-queue", attributes: .concurrent)
```

Comment:
From my understanding `DispatchQueue(label: "snail-disposer-queue")` would give you a serial queue by default. Here you're defining a concurrent queue but since you never use `async` you're not actually allowing concurrent execution.

From what I've seen elsewhere the concurrent queue should be used to perform reads using `sync` and writes using `async(flags: .barrier)` as this allows for concurrent reads but the `.barrier` enforces only one write at a time.